soc/arm/rpi_pico/rp2/soc.c

```diff
@@ -61,4 +79,22 @@ static int rp2040_init(const struct device *arg)
 	return 0;
 }
 
+/*
+ * Some pico-sdk drivers call panic on fatal error.
+ * This alternative implementation of panic handles the panic
+ * through Zephyr.
+ */
+void __attribute__((noreturn)) panic(const char *fmt, ...)
+{
+	va_list args;
+
+	va_start(args, fmt);
```

Comment:
Referencing to original PICO-SDK implementation, 
It would be better to check null-pointer here.